uint8 byte_num
uint8 mask
